test/base_test.py

```diff
@@ -1,8 +1,10 @@
 """Base class for testing with different servers"""
 
+from __future__ import annotations
+from collections.abc import Generator
```

Comment:
By curiosity, could you explain your thought process that led to the change from `typing` to `collections.abc` import ?

Reply:
Somehow I ended up on [this page](https://peps.python.org/pep-0585/) which lists all the deprecated `typing` imports.

`collections.abc` is also used in code examples in the [`typing` documentation](https://docs.python.org/3/library/typing.html#annotating-callable-objects).

Reply:
Cool, thanks for the links :slightly_smiling_face: